akka-projection-core/src/main/scala/akka/projection/ProjectionConfig.scala

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.projection
+
+import scala.concurrent.duration.FiniteDuration
+import scala.concurrent.duration._
+
+import akka.annotation.InternalApi
+import akka.util.JavaDurationConverters._
+import com.typesafe.config.Config
+
+trait ProjectionConfig {
+  def minBackoff: FiniteDuration
+  def maxBackoff: FiniteDuration
+  def randomBackoffFactor: Double
+
+  def withMinBackoff(minBackoff: FiniteDuration): ProjectionConfig
+
+  def withMaxBackoff(maxBackoff: FiniteDuration): ProjectionConfig
+
+  /**
+   * Java API
+   */
+  def withMinBackoff(minBackoff: java.time.Duration): ProjectionConfig
+
+  /**
+   * Java API
+   */
+  def withMaxBackoff(maxBackoff: java.time.Duration): ProjectionConfig
+
+  def withRandomBackoffFactor(randomBackoffFactor: Double): ProjectionConfig
+}
+
+object ProjectionConfig {
+
+  /**
+   * Java API
+   */
+  def create(): ProjectionConfig = apply()
+
+  def apply(): ProjectionConfig =
+    new ProjectionConfigImpl(3.seconds, 30.seconds, 0.2)
```

Comment:
No defaults in code. In Akka we typically have one `apply(system)` and one `apply(config)`. See one of the latest additions: https://github.com/akka/akka/blob/master/akka-actor-typed/src/main/scala/akka/actor/typed/delivery/ProducerController.scala#L177

Reply:
do we use `Option[XSettings]` to avoid forcing the user to deal with it? 

Only much later we have an `ActorSystem` that we can use to initialise the config. So, maybe the option here is to have overload methods and the `Impl` class has a `Option`. Then, if `None` and when we have already an ActorSystem available, we load the default. Otherwise we use what the user passed.

Reply:
We typically retrieve all default values from the ActorSystem or the Config parameters.
Users typically would use it as `ProjectionSettings(system).withX(...)`.

The whole ProjectionSettings parameter will be optional for the Projection, and then we load it when running the Projection.

Instead of having overloaded methods with and without the ProjectionSettings, we can have

```
CassandraProjection.atLeastOnce(....).withSettings(...)
```

Reply:
👍, this is good